lib/plug_best.ex

```diff
@@ -1,2 +1,45 @@
 defmodule PlugBest do
+  def best_language(conn = %Plug.Conn{}, supported_languages) do
+    # Fetch the raw header content
+    conn |> fetch_header_value("accept-language")
+
+    # Convert it to a list
+    |> String.split(",") |> Enum.map(&parse_header_value_item/1)
+
+    # Only keep languages that we support
+    |> Enum.filter(&(filter_header_value_item(&1, supported_languages)))
+
+    # Sort the parsed header with each score
+    |> Enum.sort(&sort_header_value_items/2)
+
+    # Return the first (best!) item
+    |> List.first
+  end
+
+  defp fetch_header_value(conn, header_name) do
+    conn |> Plug.Conn.get_req_header(header_name) |> List.first
+  end
+
+  defp parse_header_value_item(item) do
+    [language | score] = item |> String.split(";")
+
+    if Enum.empty?(score) do
```

Comment:
unassigned `if` are deprecated in Elixir 1.3.

http://tuvistavie.com/2016/elixir-1-3/#deprecation-of-imperative-assignment